index.bs

```diff
@@ -6077,12 +6077,11 @@ interface MLOperand {};
   </dl>
 </div>
 
-<div algorithm>
-To get the <dfn for="MLOperand">rank</dfn> of an {{MLOperand}} |operand|, run the following steps:
-<div class=algorithm-steps>
-    1. Return the [=list/size=] of |operand|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
-</div>
-</div>
+An {{MLOperand}}'s <dfn for=MLOperand>shape</dfn> is its {{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+
+An {{MLOperand}}'s <dfn for=MLOperand>rank</dfn> is its [=MLOperand/shape=]'s [=list/size=].
```

Comment:
{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} might not present, if MLOperand is a scalar. A scalar's rank is 0.

Reply:
Good point. Is it fair to say that the shape of a scalar is [ ] ? Or do we want to introduce an assertion, e.g. make getting the shape a sequence of steps that assert the shape exists and the size is non-zero? (and make rank smarter)

From reviewing the spec it looks like there are these cases:

* operand's descriptor's `dimensions` are used without checking -- mostly replaced here by using **shape**, so a neutral change
* operand's descriptor's `dimensions` are checked with "exists" -- that's fine

Also, MLOperand's [shape()](https://webmachinelearning.github.io/webnn/#dom-mloperand-shape) io defined as `sequence<unsigned long>` but the steps are "Return this.[[descriptor]].dimensions." - what's the intended behavior for a scalar? I'll file a bug.

Reply:
This should wait for #476 to land first, which better defines scalars.

Reply:
> Is it fair to say that the shape of a scalar is [ ] 

Correct.

> Or do we want to introduce an assertion, e.g. make getting the shape a sequence of steps that assert the shape exists and the size is non-zero?

A shape's rank can definitely be zero.

> what's the intended behavior for a scalar?

Intended behavior is a sequence of 0 elements.

Reply:
With 476's update, the dimensions should always be present and initialized to at least `[]` by default.